pkg/operator/controllers/banner/banner_controller_test.go

```diff
@@ -0,0 +1,182 @@
+			expectBanner:    true,
+			expectedMessage: "We have noticed an issue regarding your cluster requiring an action on your part. Please contact support with your cluster resource ID: FAKE_RESOURCE_ID",
+			featureFlag:     true,
+		},
+		{
+			name: "Delete existing ConsoleNotification",
+			oldCN: consolev1.ConsoleNotification{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: BannerName,
+				},
+				Spec: consolev1.ConsoleNotificationSpec{
+					Text:            "OLD BANNER TEXT",
+					Location:        consolev1.BannerTop,
+					Color:           "#000",
+					BackgroundColor: "#ff0",
+				},
+			},
+			bannerSetting: "",
+			expectBanner:  false,
+			featureFlag:   true,
+		},
+		{
+			name: "Support banner from existing ConsoleNotification",
+			oldCN: consolev1.ConsoleNotification{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: BannerName,
+				},
+				Spec: consolev1.ConsoleNotificationSpec{
+					Text:            "OLD BANNER TEXT",
+					Location:        consolev1.BannerTop,
+					Color:           "#000",
+					BackgroundColor: "#ff0",
+				},
+			},
+			bannerSetting:   "ContactSupport",
+			expectBanner:    true,
+			expectedMessage: "We have noticed an issue regarding your cluster requiring an action on your part. Please contact support with your cluster resource ID: FAKE_RESOURCE_ID",
+			featureFlag:     true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			instance := arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					ResourceID: "FAKE_RESOURCE_ID",
+					Banner: arov1alpha1.Banner{
+						Content: arov1alpha1.BannerContent(tt.bannerSetting),
+					},
+					Features: arov1alpha1.FeaturesSpec{
+						ReconcileBanner: tt.featureFlag,
+					},
+				},
+			}
+
+			r.arocli = arofake.NewSimpleClientset(&instance)
+			r.consolecli = consolefake.NewSimpleClientset(&tt.oldCN)
+
+			// function under test
+			_, err := r.Reconcile(context.Background(), ctrl.Request{})
+
+			if err != nil {
+				if tt.expectedErr == nil {
+					t.Errorf("Got unexpected Error when running Reconcile: %s", err)
+				}
+				if !strings.EqualFold(err.Error(), tt.expectedErr.Error()) {
+					t.Errorf("Expected Error %s, got %s when processing %s testcase", tt.expectedErr.Error(), err.Error(), tt.name)
+				}
+				return
+			}
+			resultBanner, err := r.consolecli.ConsoleV1().ConsoleNotifications().Get(context.Background(), BannerName, metav1.GetOptions{})
+			if tt.expectBanner {
+				if err != nil {
+					t.Error(err)
+				}
+				if !strings.EqualFold(resultBanner.Spec.Text, tt.expectedMessage) {
+					t.Error(resultBanner.Spec.Text)
+				}
+			} else {
+				if err != nil && !kerrors.IsNotFound(err) {
+					t.Errorf("Got Error when checking for empty ConsoleNotification: %s", err)
+				}
+				if err == nil || !kerrors.IsNotFound(err) || resultBanner != nil {
+					t.Errorf("Expected not to get a ConsoleNotification, but it exists")
+				}
+			}
```

Comment:
It is a nit, but I would just printing error beucase on failure it will print test name, test case name, line where it failed and the error.


```suggestion
			if err != nil {
				if tt.expectedErr == nil {
					t.Error(err)
				}
				if !strings.EqualFold(err.Error(), tt.expectedErr.Error()) {
					t.Error(err)
				}
				return
			}
			resultBanner, err := r.consolecli.ConsoleV1().ConsoleNotifications().Get(context.Background(), BannerName, metav1.GetOptions{})
			if tt.expectBanner {
				if err != nil {
					t.Error(err)
				}
				if !strings.EqualFold(resultBanner.Spec.Text, tt.expectedMessage) {
					t.Error(resultBanner.Spec.Text)
				}
			} else {
				if err != nil && !kerrors.IsNotFound(err) {
					t.Error(err)
				}
				if err == nil || !kerrors.IsNotFound(err) || resultBanner != nil {
					t.Errorf("Expected not to get a ConsoleNotification, but it exists")
				}
			}
```